db/columns/operations/select.py

```diff
@@ -133,9 +117,38 @@ def get_column_default_dict(table_oid, attnum, engine, connection_to_use=None):
     return {"value": default_value, "is_dynamic": is_dynamic}
 
 
-def get_column_default(table_oid, attnum, engine, connection_to_use=None):
+def get_column_name_from_attnum(table_oid, attnum, engine, metadata, connection_to_use=None):
+    statement = _get_columns_name_from_attnums([table_oid], [attnum], engine, metadata=metadata)
+    column_name = execute_statement(engine, statement, connection_to_use).scalar()
+    # If the column was recently dropped, it will have a name like "......pg.dropped.123....."
+    # See this example: https://stackoverflow.com/a/43050463/1714997
+    # Below line checks if the column was dropped
+    column_was_dropped = re.match(r"\.+\.pg.dropped.\d+\.\.+", column_name)
```

Comment:
If you look into the function this function calls:
```python
def _get_columns_name_from_attnums(table_oids, attnums, engine, metadata):
    pg_attribute = get_pg_catalog_table("pg_attribute", engine, metadata=metadata)
    sel = select(pg_attribute.c.attname, pg_attribute.c.attnum, pg_attribute.c.attrelid)
    conditions = [pg_attribute.c.attrelid.in_(table_oids)]
    if attnums is not None:
        conditions.append(pg_attribute.c.attnum.in_(attnums))
    else:
        conditions.extend([pg_attribute.c.attisdropped.is_(False), pg_attribute.c.attnum > 0])
    sel = sel.where(and_(*conditions))
    return sel
```
You can see where the `attisdropped` property is used, but it's inside of an `else` block. I think it should be okay to move that outside of the else block, since there's never a reason to return any column name for a dropped column.

Let me know if I need to clarify more.

Reply:
I wrote the function `_get_columns_name_from_attnums`. I agree with @mathemancer, `attisdropped.is_(False)` should always be part of the query, It was an oversight on my end